solution/0300-0399/0323.Number of Connected Components in an Undirected Graph/README_EN.md

```diff
@@ -487,4 +487,47 @@ function countComponents(n: number, edges: number[][]): number {
 
 <!-- solution:end -->
 
+<!-- solution:start -->
+
+### Solution 3: BFS
+
+<!-- tabs:start -->
+
+#### TypeScript
+
+```ts
+function countComponents(n: number, edges: number[][]): number {
+    const g: number[][] = Array.from({ length: n }, () => []);
+    const vis = new Set<number>();
+    let ans = 0;
+
+    for (const [i, j] of edges) {
+        g[i].push(j);
+        g[j].push(i);
+    }
+
+    const dfs = (i: number) => {
+        if (vis.has(i)) return;
+
+        vis.add(i);
+        for (const j of g[i]) {
+            dfs(j);
+        }
+    };
+
+    for (const i of Object.keys(g)) {
```

Comment:
In a for loop, there is no need to use Object.keys(g). Since the length of g is n, you can just use for (let i = 0; i < n; i++).

The advantage of this is that you can not use `+` to turn the number type, easier to read for newcomers.

Reply:
Nice advice, thanks a lot!
Sometimes old and time-tested solutions begin to be forgotten )

Reply:
It's nice to meet TS developers in this repository to discuss more elegant code together.    :)

Reply:
It’s mutual )

Reply:
We're both React users as well, so maybe there's a lot more to discuss in the future.

Reply:
Yes, and I am with React ecosystem since Nov. 2016, but with a few interruptions )